storage/src/test/java/org/apache/kafka/tiered/storage/integration/BaseReassignReplicaTest.java

```diff
@@ -0,0 +1,100 @@
+import java.util.Map;
+
+import static org.apache.kafka.common.utils.Utils.mkEntry;
+import static org.apache.kafka.common.utils.Utils.mkMap;
+
+public abstract class BaseReassignReplicaTest extends TieredStorageTestHarness {
+    protected final Integer broker0 = 0;
+    protected final Integer broker1 = 1;
+
+    /**
+     * Cluster of two brokers
+     * @return number of brokers in the cluster
+     */
+    @Override
+    public int brokerCount() {
+        return 2;
+    }
+
+    /**
+     * Number of partitions in the '__remote_log_metadata' topic
+     * @return number of partitions in the '__remote_log_metadata' topic
+     */
+    @Override
+    public int numRemoteLogMetadataPartitions() {
+        return 2;
+    }
+
+    @Override
+    protected void writeTestSpecifications(TieredStorageTestBuilder builder) {
+        final String topicA = "topicA";
+        final String topicB = "topicB";
+        final Integer p0 = 0;
+        final Integer partitionCount = 5;
+        final Integer replicationFactor = 2;
+        final Integer maxBatchCountPerSegment = 1;
+        final Map<Integer, List<Integer>> replicaAssignment = null;
+        final boolean enableRemoteLogStorage = true;
+        final List<Integer> metadataPartitions = new ArrayList<>();
+        for (int i = 0; i < numRemoteLogMetadataPartitions(); i++) {
+            metadataPartitions.add(i);
+        }
+
+        builder
+                // create topicA with 5 partitions, 2 RF and ensure that the user-topic-partitions are mapped to
+                // metadata partitions
+                .createTopic(topicA, partitionCount, replicationFactor, maxBatchCountPerSegment,
+                        replicaAssignment, enableRemoteLogStorage)
+                .expectUserTopicMappedToMetadataPartitions(topicA, metadataPartitions)
+                // create topicB with 1 partition and 1 RF
+                .createTopic(topicB, 1, 1, maxBatchCountPerSegment,
+                        mkMap(mkEntry(p0, Collections.singletonList(broker0))), enableRemoteLogStorage)
+                // send records to partition 0
+                .expectSegmentToBeOffloaded(broker0, topicB, p0, 0, new KeyValueSpec("k0", "v0"))
+                .expectSegmentToBeOffloaded(broker0, topicB, p0, 1, new KeyValueSpec("k1", "v1"))
+                .expectEarliestLocalOffsetInLogDirectory(topicB, p0, 2L)
+                .produce(topicB, p0, new KeyValueSpec("k0", "v0"), new KeyValueSpec("k1", "v1"),
+                        new KeyValueSpec("k2", "v2"))
+                // expand the topicB RF=2, the newly created replica gets mapped to one of the metadata partition
+                // which is being actively consumed by both the brokers
+                .reassignReplica(topicB, p0, replicaIds())
+                .expectLeader(topicB, p0, broker1, true)
+                // produce some more events and verify the earliest local offset
+                .expectEarliestLocalOffsetInLogDirectory(topicB, p0, 3L)
```

Comment:
Could you help me understand why the earliest local offset is 3, not 2 as in L80?

Reply:
We are producing one more event (k3, v3) in L89 and expect the earliest local log to move to offset 3.

Reply:
Oh, right, I missed that. Thanks.